server/controllers/accounts.controller.js

```diff
@@ -9,52 +12,76 @@ const AccountsController = {
    *
    * @param  {object} res - response
    *
-   * @return {json} - jsonObject containing status code and or error
+   * @return {json} - jsonObject containing status code and data or error
    *
    * Route: POST: /accounts
    *
    * */
   async create(req, res) {
     const { error, isValid } = validateCreateBankAccountInput(req.body);
     if (!isValid) {
-      return res.status(400).json({
-        status: 400,
-        error,
-      });
+      return res.status(400).json({ status: 400, error });
     }
 
-    const newAccount = await Account.create(req.body);
-    return res.status(201).json({
-      status: 201,
-      data: {
-        accountNumber: newAccount.accountNumber,
-        firstName: newAccount.firstName,
-        lastName: newAccount.lastName,
-        email: newAccount.email,
-        type: newAccount.type,
-        openingBalance: newAccount.balance,
-      },
-    });
+    const {
+      firstName, lastName, email, type,
+    } = req.body;
+
+    // Generate account account number
+    const date = new Date();
+    const accountNumber = date.getTime();
+
+    // Check if user has a registered account
+    User.findOne('email', email)
+      .then((result) => {
+        if (!result || result.rows.length < 1) {
+          return res.status(400).json({
+            status: 400,
+            error: 'Please create a user account to continue',
+          });
+        }
+        const accountDetail = {
+          account_number: accountNumber,
+          owner: result.rows[0].id,
+          type,
+        };
+
+        // Save account
+        return Account.save(accountDetail).then(data => data.rows[0])
+          .then(account => res.status(201).json({
+            status: 201,
+            data: {
+              accountNumber: account.account_number,
+              firstName,
+              lastName,
+              email,
+              type: account.type,
+              openingBalance: account.balance,
+            },
+          }))
+          .catch(err => res.status(400).json({
+            status: 400,
+            error: `Could not create account ${err}`,
```

Comment:
Change status code here too.

Reply:
I changed this to server error(500). Is it good?

Reply:
Please where is the change? I can't see it.